### ISSUE TITLE
Minimization failed after 100 attempts

### ISSUE THREAD

**Issue by Charlesjc-lab** [YuzheWangPKU/DiffPepBuilder#1]
In the redocking step, I encounter an error message that says "Minimization failed after 100 attempts." Can the author provide a  solution?

**Comment by YuzheWangPKU** [comment]
Please try the latest version which includes updates to the redocking pipeline. If the problem persists, we recommend manually inspecting and removing any structures with backbone steric clashes, as these may cause the energy minimization process to fail.

We have empirically observed that including too many residues as hotspots or motifs may bias the model towards generating peptides with significant steric clashes. If you find that many peptides are exhibiting steric clashes, please consider adjusting the binding information provided to the model accordingly.

**Comment by Charlesjc-lab** [comment]
Yes, I have also tried the method you mentioned, using the new scripts and manual check to solve similar problems. I also have suggestions and ideas for generating conformations, and it is possible to filter the output structures using the af2 method in dl_binder_design/colabdesign. However, unfortunately, AF2 is not a good filtering method for diffpepbuilder. Are there any effective filtering methods that the author has actually used?

**Comment by wfh1998** [comment]
Thank you for your suggestion. We previously attempted to perform redocking tests on several peptide structures using AF2 (as we are also working on peptide docking). The results indicated that AF2 performs poorly in predicting certain loop conformations of peptides in redocking tests. Therefore, we opted to use a relax protocol based on Rosetta's energy score for post-processing. However, the method you mentioned, such as dlbinderdesign, is also a reasonable approach for subsequent backbone and sequence optimization pepline, and it is worth exploring.

**Comment by wfh1998** [comment]
> Yes, I have also tried the method you mentioned, using the new scripts and manual check to solve similar problems. I also have suggestions and ideas for generating conformations, and it is possible to filter the output structures using the af2 method in dl_binder_design/colabdesign. However, unfortunately, AF2 is not a good filtering method for diffpepbuilder. Are there any effective filtering methods that the author has actually used?


Thank you for your suggestion. We previously attempted to perform redocking tests on several peptide structures using AF2 (as we are also working on peptide docking). The results indicated that AF2 performs poorly in predicting certain loop conformations of peptides in redocking tests. Therefore, we opted to use a relax protocol based on Rosetta's energy score for post-processing. However, the method you mentioned, such as dlbinderdesign, is also a reasonable approach for subsequent backbone and sequence optimization pepline, and it is worth exploring.

**Comment by amin-sagar** [comment]
Hello. Sorry for commenting on a closed issue.
I am also observing structures with significant backbone clashes.
Do you have some general recommendations on the number of hotspot residues, surface area of the selected pocket etc., based on your experience?
Best,
Amin.

**Comment by Charlesjc-lab** [comment]
When employing an input akin to RFdiffsion (a hotspot comprising 3 to 6 hydrophobic residues), results can be obtained. However, it is challenging to circumvent instances of conflict. I wish it is worth to you.

Furthermore, when employing a threshold value of T=0.001, the number of staple peptides identified is significantly reduced.

**Comment by YuzheWangPKU** [comment]
Thanks for the valuable comment from @Charlesjc-lab! In our experiments, we used alanine scanning to identify hotspot residues for the three targets (3CLpro, ALK1, TNF-α). We found this approach effective for our model. For targets that lack a native binding ligand or natural substrate, we acknowledge that iterative optimization of the hotspot residues may still be necessary, and we found the hotspot selection procedures from RFdiffusion to be beneficial.

If significant clashes persist, an alternative to consider is manually adjusting the `pocket_cutoff` parameter in the script `experiments/process_receptor.py` using the flag `--pocket_cutoff`. Our previous experiments indicated that an excessively large pocket can lead to generated peptide binders becoming overly "deep" within the pocket, leading to the clashes.

**Comment by amin-sagar** [comment]
Thanks @YuzheWangPKU and @Charlesjc-lab 
Does it make sense to include a structure quality estimate like Qmean in the workflow and filter out the structures with bad geometry? Especially, since DiffPepBuilder is quick fast in generating peptides.

**Comment by YuzheWangPKU** [comment]
> Thanks @YuzheWangPKU and @Charlesjc-lab Does it make sense to include a structure quality estimate like Qmean in the workflow and filter out the structures with bad geometry? Especially, since DiffPepBuilder is quick fast in generating peptides.

Thank you for your suggestion! In our design pipeline, we primarily use Rosetta ddG as a criterion to filter out generated peptides that exhibit backbone clashes or distorted structures, and we have found this approach to be effective. We plan to integrate it into our pipeline in a more end-to-end manner in the next version.

**Comment by amin-sagar** [comment]
Thanks.

My binding site is quite charged. I had specified 6 residues as hotspot but none were hydrophobic. I am wondering if this is leading to peptides with backbone clashes.

For the three targets that you studied, did you always choose hydrophobic residues as your hotspots? Or did you have combination of hydrophobic and charged residues?

Thanks again.
Amin.

**Comment by YuzheWangPKU** [comment]
Hi @amin-sagar, many apologies for the delayed response—I just realized I had missed this thread entirely. Our model uses the specified hotspot residues primarily to define the binding pocket, so whether they are hydrophobic or charged shouldn’t significantly affect backbone clashes. I’d recommend experimenting with different combinations and positions of hotspot residues to help mitigate the issue. Hope this helps, and thanks again for your patience!
